src/ext_slice.rs

```diff
@@ -429,6 +430,42 @@ pub trait ByteSlice: Sealed {
         }
     }
 
+    /// Write a UTF-8 debug representation of a potentially UTF-8 invalid
+    /// sequence.
+    ///
+    /// This method encodes a bytes slice into a UTF-8 valid representation by
+    /// writing invalid sequences as `\xXX` escape codes.
+    ///
+    /// This method uses `char::escape_debug` which means UTF-8 valid characters
+    /// like `\n` and `\t` are also escaped.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use bstr::ByteSlice;
+    ///
+    /// let mut message = String::from("cannot load such file -- ");
+    /// let filename = b"utf8-invalid-name-\xFF";
+    /// filename.escape_debug_into(&mut message).unwrap();
+    /// assert_eq!(r"cannot load such file -- utf8-invalid-name-\xFF", message);
+    /// ```
+    #[inline]
+    fn escape_debug_into<T: fmt::Write>(&self, f: &mut T) -> fmt::Result {
+        let buf = self.as_bstr();
```

Comment:
Why are you converting the slice to a `&BStr` first? It appears unnecessary and also results in an extra `as_bytes()` call below.

Reply:
this was a holdover from my own implementation of this function in my downstream project. I didn't realize all these functions are implemented on `&[u8]` too. fixed.